src/index.ts

```diff
@@ -1,14 +1,41 @@
 import express from 'express';
+import logRequest from './middlewares/logRequest';
+import GenerateService from './services/generate';
 
 //TODO: replace with injectable / importable config object
 const config = {
     port: process.env.PORT || 4000,
+    articleStorePath: process.env.ARTICLE_STORE_PATH || 'http://localhost:8080/articles/',
 };
 
 console.log('Starting server...');
 
+// Initialize services
+const generationService = GenerateService(config);
+
 const app: express.Application = express();
+
+// Log all requests on this route.
+app.use(logRequest);
+
 app.get('/health', (_, res) => res.sendStatus(200));
+app.post('/generate/:articleId', async (req, res) => {
+    try {
+        await generationService.startJob(req.params.articleId);
+    } catch (error) {
+        switch (error.message) {
```

Comment:
Not sure if this is right or not. I've never been a fan of needing to pass the `resp` to a service function. This keeps the layers separate making it easier to test but still doesn't quite feel right 🤔 .

Reply:
This looks OK to me, I can't from a quick look see why you'd need/want to pass the response down.